sdktests/custom_matchers_contexts.go

```diff
@@ -0,0 +1,109 @@
+package sdktests
+
+import (
+	h "github.com/launchdarkly/sdk-test-harness/v2/framework/helpers"
+
+	"github.com/launchdarkly/go-sdk-common/v3/ldcontext"
+	"github.com/launchdarkly/go-sdk-common/v3/ldvalue"
+	m "github.com/launchdarkly/go-test-helpers/v2/matchers"
+)
+
+// JSONMatchesContext builds a Matcher to verify that the input JSON is a valid representation of the
+// specified context. This is using the regular context schema (i.e. what would be sent to evaluation
+// endpoints), not the event schema.
+//
+// The matcher should be tolerant of all allowable variants: for instance, it is legal to include
+// `"transient": false` in the representation rather than omitting transient.
```

Comment:
The logic for this is a bit convoluted, but I think it's worthwhile so that we don't have to embed logic like this throughout various tests. There are unit tests in the next file to verify the behavior, so as long as those are covering enough valid and invalid input combinations, we should be able to rely on this matcher.